### ISSUE TITLE
Support for subproject wrap files

### ISSUE THREAD

**Issue by realmazharhussain** [mesonbuild/vscode-meson#150]
It would be great if the extension also provided syntax highlighting and completion for meson wrap files.

**Comment by tristan957** [comment]
For syntax highlighting, use `"files.associations": {"*.wrap": "ini"}`

**Comment by xclaesse** [comment]
Would be nice if vscode extension can add that association automatically. Is it possible?

**Comment by realmazharhussain** [comment]
> For syntax highlighting, use `"files.associations": {"*.wrap": "ini"}`

Thank you for the tip. But I was asking for actual support for those files. e.g. If I mistype something, it shows squiggles.

The default ini style does not differentiate between keys that are meaningful in a wrap file and the ones that are not.

Even just a different color for the keys that are actually supported by meson would be a big step.

**Comment by xclaesse** [comment]
> It could be, but honestly I hate when extensions change my settings.

True, maybe it's possible to be more specific like `subprojects/*.wrap` ?

**Comment by tristan957** [comment]
There is a bug in VSCode that doesn't allow that if I remember correctly

I think `**/subprojects/*.wrap` works though.

**Comment by shanoaice** [comment]
> > For syntax highlighting, use `"files.associations": {"*.wrap": "ini"}`
> 
> Thank you for the tip. However, I was asking for actual support for those files. e.g. If I mistype something, it shows squiggles. The default ini style does not even differentiate between keys that are meaningful in a wrap file and the ones that are not.
> 
> Even just a different color for the keys that are actually supported by meson would be a big step.

Maybe you should ask for muon or Swift-MesonLSP developers for the corresponding support? After all vscode-meson is using them for completion and linting.
